From 06a9806dc16c7165379b3ca962dab5b18a5bae31 Mon Sep 17 00:00:00 2001
From: xdnax <mikecoconut5832@gmail.com>
Date: Fri, 23 Dec 2016 09:52:57 -0500
Subject: [PATCH 14/15] Expose External Settings Icon Tint

Allows themers to use a custom color to tint Settings's icons like the
"G" icon. It breaks the color from Settings's accent and defines its
own... Pixel Blue ROM Devs may need to change the reference in
colors.xml so it gets themed the correct color.

Change-Id: Ifb4a99a315271cdcf978a796e3dea0eaef8d7750
---
 res/values/projekt_colors.xml                            | 6 +++---
 src/com/android/settings/dashboard/DashboardAdapter.java | 5 +----
 2 files changed, 4 insertions(+), 7 deletions(-)

diff --git a/res/values/projekt_colors.xml b/res/values/projekt_colors.xml
index d131ecfc1..b960e8867 100644
--- a/res/values/projekt_colors.xml
+++ b/res/values/projekt_colors.xml
@@ -15,6 +15,7 @@
     limitations under the License.
 -->
 <resources>
+
     <!-- Dashboard category and tile background -->
     <color name="dashboard_category_title_background">@color/card_background</color>
     <color name="selectable_card_background">@color/card_background</color>
@@ -39,19 +40,18 @@
     <color name="storage_private_3">#fff2a600</color>
     <color name="storage_private_4">#ffec407a</color>
     <color name="storage_private_5">#ffc0ca33</color>
-
     <!-- Linear color bar -->
     <color name="linear_color_bar_left">@*android:color/material_deep_teal_500</color>
     <color name="linear_color_bar_middle">@*android:color/material_deep_teal_500</color>
     <color name="linear_color_bar_right">#ffced7db</color>
-
     <!-- External Icon Tint -->
     <color name="external_tile_icon_tint_color">?android:attr/colorAccent</color>
     <!-- Settings switchbar background -->
     <color name="switchbar_background_color">?android:attr/colorSecondary</color>
-
     <!-- Bluetooth pairing pin confirm dialog -->
     <color name="bluetooth_pairing_caption">@*android:color/secondary_text_material_light</color>
     <color name="bluetooth_pairing_code_message">@*android:color/secondary_text_material_light</color>
     <color name="bluetooth_phonebook_sharing_message_confirm_pin">@*android:color/secondary_text_material_light</color>
+    <!-- External Icon Tint -->
+    <color name="external_icon_tint">@*android:color/accent_device_default_light</color>
 </resources>
diff --git a/src/com/android/settings/dashboard/DashboardAdapter.java b/src/com/android/settings/dashboard/DashboardAdapter.java
index ba7c4185b..6b2ac30c8 100644
--- a/src/com/android/settings/dashboard/DashboardAdapter.java
+++ b/src/com/android/settings/dashboard/DashboardAdapter.java
@@ -120,9 +120,6 @@ public class DashboardAdapter extends RecyclerView.Adapter<DashboardAdapter.Dash
         mCategories = categories;
 
         // TODO: Better place for tinting?
-        TypedValue tintColor = new TypedValue();
-        mContext.getTheme().resolveAttribute(com.android.internal.R.attr.colorAccent,
-                tintColor, true);
         for (int i = 0; i < categories.size(); i++) {
             for (int j = 0; j < categories.get(i).tiles.size(); j++) {
                 Tile tile = categories.get(i).tiles.get(j);
@@ -131,7 +128,7 @@ public class DashboardAdapter extends RecyclerView.Adapter<DashboardAdapter.Dash
                         tile.intent.getComponent().getPackageName())) {
                     // If this drawable is coming from outside Settings, tint it to match the
                     // color.
-                    tile.icon.setTint(tintColor.data);
+                    tile.icon.setTint(mContext.getColor(R.color.external_icon_tint));
                 }
             }
         }
-- 
2.11.1

